mkosi/__init__.py

```diff
@@ -8002,20 +8003,31 @@ def expand_paths(paths: List[str]) -> List[str]:
     expanded = []
     for path in paths:
         try:
-            expanded += [string.Template(path).substitute(environ)]
+            expanded += [Path(string.Template(path).substitute(environ))]
         except KeyError:
             # Skip path if it uses a variable not defined.
             pass
     return expanded
 
 
-def prepend_to_environ_path(paths: List[Path]) -> None:
+@contextlib.contextmanager
+def prepend_to_environ_path(paths: Sequence[Path]) -> Iterator[None]:
     if not paths:
         return
```

Comment:
The `with` statement in `run_verb` doesn't like the early exit here. You can put the whole body of the with statement in this function except for the yield behind this conditional, then it should work fine.

Reply:
I just added an extra `yield` before, that also works